File: scala-kata/src/main/scala/org/suggs/katas/bankaccount/Money.scala
package org.suggs.katas.bankaccount

object Money {
  def anAmountOf(amount: Double) = new Money(amount)
}

case class Money(amount: Double) {

  def less(subtraction: Money): Money = Money(amount - subtraction.amount)

  def add(addition: Money): Money = Money(amount + addition.amount)
}

File: java-kata/src/main/java/org/suggs/katas/bankaccount/Money.java
package org.suggs.katas.bankaccount;

import lombok.EqualsAndHashCode;

@EqualsAndHashCode
public class Money {
    private double amount;

    private Money(final double anAmount) {
        this.amount = anAmount;
    }

    public static Money anAmountOf(final double anAmount) {
        return new Money(anAmount);
    }

    public void add(final Money anAmount) {
        amount = this.amount + anAmount.amount;
    }

    public void less(final Money anAmount) {
        amount = this.amount - anAmount.amount;
    }
}
